### ISSUE TITLE
CMU MOSEI labels

### ISSUE THREAD

**Issue by EmileChapuis** [A2Zadeh/CMU-MultimodalSDK#241]
Hi, 

First of all thank you for your work. I wanted to know if you valid the data from http://immortal.multicomp.cs.cmu.edu/raw_datasets/processed_data/cmu-mosei/seq_length_20/data/. If you do, can you tell me more about emotion labels : there are a lot of null rows and thus number of labels reported in the paper don't fit. Moreover how did you deal with cases where several labels had the same Likert scale (e.g., [1,1,0,0,0,0])

Regards

**Comment by A2Zadeh** [comment]
Hi @EmileChapuis,

It would be better to use the seq_length_50 data. It is the more updated and better documented one. It belongs to the Multimodal Transformer by Hubert Tsai. The version you are working with is the original one that the paper used, but there are lots of inter-dependencies between data and code. The 50 seq length is better documented and more up to date, you can trim it to 20 and basically get the same as the seq_length_20. Does that address your question?

**Comment by AnilRahate** [comment]
Hi @A2Zadeh 
Can you please guide how to deal with emotion labels as it contains multiple labels and that too integer values for examples. I am thinking how to apply softmax in output layer for such cases.

**Comment by A2Zadeh** [comment]
Hi @AnilRahate 

Multiple emotions can be present at the same time in CMU-MOSEI. Therefore, you should not use softmax (IEMOCAP style). It should be a softmax on each emotion, or multiclass classification.

**Comment by AnilRahate** [comment]
@A2Zadeh Thanks for immediate response. l will try to understand this more. 
One more question, for Sequence length 50, there are  3 files, will mosei_senti_data.pkl file provide aligned data for  emotion classification. Also how to read .dt file which is 3rd file in same location. http://immortal.multicomp.cs.cmu.edu/raw_datasets/processed_data/cmu-mosei/seq_length_50/

**Comment by EmileChapuis** [comment]
Thanks for your response I will have a look to Hubert Tsai data, but it seems only annotated in sentiment. I understand that it can have some dependencies but still there are 13969 null rows (no emotion detected: [0,0,0,0,0,0]) out of 22413 rows (counting train, valid and test split) and {'Happy': 5159, 'Anger': 1596, 'Sad': 1113, 'Disgust': 1059, 'Surprise': 204, 'Fear': 176}. Can you provide use some code snippet or a clue on how you manage to fit the emotions distribution in the paper ?

Regards,

**Comment by AnilRahate** [comment]
@EmileChapuis, are you using data from this link http://immortal.multicomp.cs.cmu.edu/raw_datasets/processed_data/cmu-mosei/seq_length_20/data/.  I found total 3322 rows with all zeros [ ey_train - 2243, ey_valid - 345 & ey_test - 734]. Am I missing some thing. Total size of ey_train:15290, ey_valid:2291, ey_test:4832
